drivers/led/is31fl3733.c

```diff
@@ -0,0 +1,298 @@
+	uint8_t scratch_buf[IS31FL3733_MAX_LED + 1];
+	/* LED config reg state, IS31FL3733 conf reg is write only */
+	uint8_t conf_reg;
+};
+
+/* Selects target register page for IS31FL3733. After setting the
+ * target page, all I2C writes will use the selected page until the selected
+ * page is changed.
+ */
+static int is31fl3733_select_page(const struct device *dev, uint8_t page)
+{
+	const struct is31fl3733_config *config = dev->config;
+	struct is31fl3733_data *data = dev->data;
+	int ret = 0U;
+
+	if (data->selected_page == page) {
+		/* No change necessary */
+		return 0;
+	}
+
+	/* Unlock page selection register */
+	ret = i2c_reg_write_byte_dt(&config->bus, CMD_LOCK_REG, CMD_LOCK_UNLOCK);
+	if (ret < 0) {
+		LOG_ERR("Could not unlock page selection register");
+		return ret;
+	}
+
+	/* Write to function select to select active page */
+	ret = i2c_reg_write_byte_dt(&config->bus, CMD_SEL_REG, page);
+	if (ret < 0) {
+		LOG_ERR("Could not select active page");
+		return ret;
+	}
+	data->selected_page = page;
+	return ret;
+}
+
+static int is31fl3733_led_set_brightness(const struct device *dev, uint32_t led, uint8_t value)
+{
+	const struct is31fl3733_config *config = dev->config;
+	int ret;
+	uint8_t led_brightness = (uint8_t)(((uint32_t)value * 255) / 100);
+
+	if (led >= IS31FL3733_MAX_LED) {
+		return -EINVAL;
+	}
+
+	/* Configure PWM mode */
+	ret = is31fl3733_select_page(dev, CMD_SEL_PWM);
+	if (ret < 0) {
+		return ret;
+	}
+	return i2c_reg_write_byte_dt(&config->bus, led, led_brightness);
+}
+
+static int is31fl3733_led_on(const struct device *dev, uint32_t led)
+{
+	return is31fl3733_led_set_brightness(dev, led, IS31FL3733_MAX_BRIGHTNESS);
+}
+
+static int is31fl3733_led_off(const struct device *dev, uint32_t led)
+{
+	return is31fl3733_led_set_brightness(dev, led, 0);
+}
+
+static int is31fl3733_led_write_channels(const struct device *dev, uint32_t start_channel,
+					 uint32_t num_channels, const uint8_t *buf)
+{
+	const struct is31fl3733_config *config = dev->config;
+	struct is31fl3733_data *data = dev->data;
+	int ret = 0U;
+	uint8_t *pwm_start;
+
+	if ((start_channel + num_channels) > IS31FL3733_MAX_LED) {
+		return -EINVAL;
+	}
+	pwm_start = data->scratch_buf + start_channel;
+	/* Set PWM and LED target registers as first byte of each transfer */
+	*pwm_start = start_channel;
+	memcpy((pwm_start + 1), buf, num_channels);
+
+	/* Write LED PWM states */
+	ret = is31fl3733_select_page(dev, CMD_SEL_PWM);
+	if (ret < 0) {
+		return ret;
+	}
+	LOG_HEXDUMP_DBG(pwm_start, (num_channels + 1), "PWM states");
+	return i2c_write_dt(&config->bus, pwm_start, num_channels + 1);
+}
+
+static int is31fl3733_init(const struct device *dev)
+{
+	const struct is31fl3733_config *config = dev->config;
+	struct is31fl3733_data *data = dev->data;
+	int ret = 0U;
+	uint8_t dummy;
+
+	if (!i2c_is_ready_dt(&config->bus)) {
+		LOG_ERR("I2C device not ready");
+		return -ENODEV;
+	}
+	if ((config->sdb.port != NULL) && !gpio_is_ready_dt(&config->sdb)) {
```

Comment:
What do you think about that?

<pre>
if ((config->sdb.port != NULL) { 
    if (!gpio_is_ready_dt(&config->sdb)) {
        LOG_ERR("GPIO SDB pin not ready");
        return -ENODEV;
    }
    ret = gpio_pin_configure_dt(&config->sdb, GPIO_OUTPUT_ACTIVE);
    if (ret < 0) {
	return ret;
    }
}

Reply:
Sure, this is functionally equivalent based on the device datasheet. Changed code to this control flow.